sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -62,8 +61,7 @@ public String build(Expression expr) {
       String name = e.name();
       switch (name) {
         case "IN": {
-          List<String> children =
-            Arrays.stream(e.children()).map(c -> build(c)).collect(Collectors.toList());
+          List<String> children = expressionsToStringList(e.children());
```

Comment:
nit: Given this is called only here, by not avoid the subList ? (give start and end params to `expressionsToStringList`)

Reply:
is [e00330f](https://github.com/apache/spark/pull/37843/commits/e00330fd2c719406d189bbfe49468bcabca76ea0) ok ?

Reply:
But the `.subList` only wraps a `SubList` object and does not trigger operations such as memory copy,  so the performance gap may be small

Reply:
Agree, sublist is fairly optimal ... but can be avoided here is possible. It is a nit comment actually :-)